### PR TITLE
Update merkletree behaviour

### LINKED ISSUE #979: Update merkletree behaviour
- Update hash of the bytecode:
  - spec: https://hackmd.io/bbknowuiQxm3xrgWOt890g#SC-Code
  - impl: https://github.com/0xPolygonHermez/zkevm-commonjs/blob/main/src/smt-utils.js#L403
- Add bytecode length leaf
  - spec: https://hackmd.io/bbknowuiQxm3xrgWOt890g#SC-length

Note that this is important for the genesis, as other smart contracts will be deployed through transactions and the executor will handle this

### PR DESCRIPTION
**Pull request by ToniRamirezM**
Closes #979 

### What does this PR do?

- Implements new leaf type (code lenght) and new byte code hash function.
- Adds tests for this.

### Reviewers

Main reviewers:

- @arnaubennassar 
- @fgimenez 

Codeowner reviewers:

- @tclemos

### CODE REVIEW

File: state/state_test.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func TestMain(m *testing.M) {
 	}
 	defer stateDb.Close()
 
-	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "localhost")
+	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "54.170.178.97")
```
Comment (fgimenez) [inline state/state_test.go:62]: We should keep localhost here
```suggestion
	zkProverURI := testutils.GetEnv("ZKPROVER_URI", "localhost")
```

File: state/state_test.go [hunk 2]
```diff
@@ -1226,3 +1226,85 @@ func TestExecutorTxHash(t *testing.T) {
...
+				Type:            int(merkletree.LeafTypeStorage),
+				StoragePosition: "0x0000000000000000000000000000000000000000000000000000000000005BBF",
+				Value:           "0x0B6E",
+			},
+		},
+	}
+
+	if err := dbutils.InitOrReset(cfg); err != nil {
+		panic(err)
+	}
```
Comment (fgimenez) [inline state/state_test.go:1300]:
```suggestion
	require.NoError(t, dbutils.InitOrReset(cfg))
```

**Review by fgimenez — APPROVED** [review]
:rocket:

File: state/state_test.go (hunk 1 above)
Comment (ToniRamirezM) [reply state/state_test.go:62]: Yes, I should have not pushed that

**Review by tclemos — APPROVED** [review]

**Pull request merged**